base/genimport/importer.go

```diff
@@ -269,14 +272,120 @@ func removeAllFilesInDirExcept(o *Output, dir string, except_list []string) {
 }
 
 func createPluginGoModFile(o *Output, pkgpath string, dir string) string {
+	newLines := [][]byte{
+		[]byte("module gomacro.imports/" + pkgpath + "\n"),
+	}
+
+	if localpath, ok := local(o, pkgpath); ok {
+		newLines = append(newLines, goModReplacementDirectives(o, localpath)...)
+	}
+
+	newGoModText := bytes.Join(newLines, nil)
+
 	gomod := paths.Subdir(dir, "go.mod")
-	err := ioutil.WriteFile(gomod, []byte("module gomacro.imports/"+pkgpath+"\n"), os.FileMode(0644))
+	err := ioutil.WriteFile(gomod, newGoModText, os.FileMode(0o644))
 	if err != nil {
 		o.Errorf("error writing file %q: %v", gomod, err)
 	}
+
 	return gomod
 }
 
+// local returns a localized path for the package if it is local, as well as a
+// boolean indicating if it was found
+func local(o *Output, pkgpath string) (string, bool) {
+	cwd, err := os.Getwd()
+	if err != nil {
+		o.Errorf("error finding cwd: %v", err)
+		return "", false
+	}
+	// Hacky way of determining if we're running gomacro from a directory that
+	// likely shares a go.mod file with the pkgpath. It assumes the both the
+	// cwd and module location will include the pattern 'owner/project', which
+	// is often true.
+	// TODO: find a better way to determine if it's local
+	modParts := strings.Split(pkgpath, "/")
+	if len(modParts) > 2 &&
+		strings.Contains(strings.ToLower(cwd), strings.ToLower(strings.Join(modParts[1:3], "/"))) {
+		return cwd, true
+	}
+
+	return "", false
+}
+
+// goModReplacementDirectives will create the replacement directives associated
+// with a module that can be found locally, for the purpose of use in a
+// gomacro.imports mod file.
+func goModReplacementDirectives(o *Output, localpath string) [][]byte {
+	goModDir, goModFile, err := findGoMod(localpath)
+	if err != nil {
```

Comment:
if this is `noGoModOnPath`, we probably want to just return null without warning -- that's the case where you're in some random directory, not trying to do a local import (and indeed we should in that case make sure we just assume your import is not in fact local!)

Reply:
Hm. I think the case you're specifying is handled by the boolean check on `local`? https://github.com/Khan/gomacro/blob/e55ee175015726079e5fa4d3fab146fa60748c7a/base/genimport/importer.go#L279

On the assumption that we're only in this code if we think we have a valid local path to the package, I feel like it should still probably complain if it doesn't find a `go.mod` here, otherwise the behavior might be very confusing. Although I wonder if we alternately report in `local` that we have (and/or possibly have not) found the path locally?

Either way, I should probably update the method doc to suggest that.